contracts/ronin/validator/CoinbaseExecution.sol

```diff
@@ -404,26 +404,25 @@ abstract contract CoinbaseExecution is
     uint256 _newValidatorCount,
     uint256 _newPeriod
   ) private {
-    for (uint256 _i = _newValidatorCount; _i < validatorCount; _i++) {
+    // Remove exceeding validators in the current set
+    for (uint256 _i = 0; _i < validatorCount; _i++) {
       delete _validatorMap[_validators[_i]];
       delete _validators[_i];
     }
 
-    uint256 _count;
+    // Remove flag for all validator in the current set
+    for (uint _i = 0; _i < _newValidatorCount; _i++) {
+      delete _validatorMap[_validators[_i]];
+    }
```

Comment:
Is this already done by the previous loop?

Reply:
The previous loop should be `for (uint256 _i = _newValidatorCount; _i < validatorCount; _i++)` instead.

Reply:
Will the variable `validatorMap` be reassigned at the next loop?